src/features/tags/hooks/useTag.ts

```diff
@@ -0,0 +1,53 @@
+import { useSelector, useStore } from 'react-redux';
+
+import { loadItemIfNecessary } from 'core/caching/cacheUtils';
+import { RootState } from 'core/store';
+import { ZetkinTag } from 'utils/types/zetkin';
+import { tagAssigned, tagLoad, tagLoaded, tagUnassigned } from '../store';
+import { useApiClient, useEnv } from 'core/hooks';
+
+interface UseTagReturn {
+  assignToPerson: (personId: number, value?: string) => void;
+  removeFromPerson: (personId: number) => Promise<void>;
+  tagData: ZetkinTag | null;
+  tagError: unknown | null;
+  tagIsLoading: boolean;
+}
+export default function useTag(orgId: number, tagId: number): UseTagReturn {
+  const apiClient = useApiClient();
+  const tags = useSelector((state: RootState) => state.tags);
+  const env = useEnv();
+  const store = useStore();
+
+  const assignToPerson = async (personId: number, value?: string) => {
+    const data = value ? { value } : undefined;
+    const tag = await apiClient.put<ZetkinTag>(
+      `/api/orgs/${orgId}/people/${personId}/tags/${tagId}`,
+      data
+    );
+    store.dispatch(tagAssigned([personId, tag]));
+  };
+  const getTag = () => {
```

Comment:
This function is unnecessary. Inline it's contents inside the hook instead. You can just grab the return value from `loadItemIfNecessary` and use that further down the function where you are currently calling `getTag()`.